Sources/Purchasing/OfferingsManager.swift

```diff
@@ -111,25 +127,53 @@ class OfferingsManager {
 
 private extension OfferingsManager {
 
-    func handleOfferingsBackendResult(
-        with response: OfferingsResponse,
+    func fetchCachedOfferingsFromDisk(
+        appUserID: String,
         fetchPolicy: FetchPolicy,
-        completion: (@MainActor @Sendable (Result<Offerings, Error>) -> Void)?
+        completion: (@escaping @Sendable (Offerings?) -> Void)
+    ) {
+        guard let data = self.deviceCache.cachedOfferingsData(appUserID: appUserID),
+              let response: OfferingsResponse = try? JSONDecoder.default.decode(jsonData: data) else {
+            completion(nil)
+            return
+        }
+
+        self.createOfferings(
+            from: response,
+            fetchPolicy: fetchPolicy,
+            completion: { [cache = self.deviceCache] result in
+                switch result {
+                case let .success(offerings):
+                    Logger.debug(Strings.offering.vending_offerings_cache_from_disk)
+
+                    cache.cacheInMemory(offerings: offerings)
+                    completion(offerings)
+
+                case .failure:
+                    completion(nil)
+                }
+            }
+        )
+    }
+
+    func createOfferings(
```

Comment:
Extracted this to be used from both `fetchCachedOfferingsFromDisk` and `handleOfferingsBackendResult`.